src/components/pagination/pagination.js

```diff
@@ -0,0 +1,143 @@
+import '@brightspace-ui/core/components/button/button-icon.js';
+import '@brightspace-ui/core/components/inputs/input-number.js';
+import '@brightspace-ui/core/components/inputs/input-text.js';
+import { css, html, LitElement } from 'lit';
+import { LocalizeLabsElement } from '../localize-labs-element.js';
+import { RtlMixin } from '@brightspace-ui/core/mixins/rtl-mixin.js';
+import { selectStyles } from '@brightspace-ui/core/components/inputs/input-select-styles.js';
+
+class Pagination extends RtlMixin(LocalizeLabsElement(LitElement)) {
+
+	static get properties() {
+		return {
+			pageNumber: { type: Number, attribute: 'page-number', reflect: true },
+			maxPageNumber: { type: Number, attribute: 'max-page-number', reflect: true },
+			showItemCountSelect : { type: Boolean, attribute: 'show-item-count-select', reflect: true },
+			itemCountOptions : { type: Array, attribute: 'item-count-options' },
+			selectedCountOption : { type: Number, attribute: 'selected-count-option', reflect: true },
+		};
+	}
+
+	static get styles() {
+		return [selectStyles, css`
+			:host {
+				align-items: center;
+				display: flex;
+				justify-content: center;
+				white-space: nowrap;
+			}
+
+			.d2l-page-selector-container {
+				margin: 0.75rem;
+			}
+
+			.d2l-page-number {
+				margin-left: 0.25rem;
+				margin-right: 0.25rem;
+				width: 4rem;
+			}
+
+			.d2l-page-max-text {
+				margin-right: 0.25rem;
+			}
+
+			@media (max-width: 544px) {
+				:host {
+					flex-direction: column-reverse;
+				}
+			}
+		`];
+	}
+
+	constructor() {
+		super();
+		this.pageNumber = 1;
+		this.maxPageNumber = 1;
+		this.itemCountOptions = [10, 20, 30, 40];
+	}
+
+	render() {
+		return html`
+			<div class="d2l-page-selector-container">
+				<d2l-button-icon id="d2l-labs-pagination-previous-button" icon="tier1:chevron-left" @click="${this._onPreviousClicked}" text="${this.localize('components:pagination:previousPage')}" ?disabled=${this._isFirstPage()}></d2l-button-icon>
+
+				<d2l-input-number
+					class="d2l-page-number"
+					autocomplete="off"
+					label="${this.localize('components:pagination:currentPage', { pageNumber: this.pageNumber, maxPageNumber: this.maxPageNumber })}"
+					label-hidden
+					max-fraction-digits="0"
+					value="${this.pageNumber}"
+					@change=${this._onPageNumberChanged}>
+				</d2l-input-number>
+				<!-- Note: this uses a division slash rather than a regular slash -->
+				<!-- a11y note: setting aria-hidden to true because info here is covered by input element -->
+				<span class="d2l-page-max-text" aria-hidden="true">∕ ${this.maxPageNumber}</span>
+				<d2l-button-icon id="d2l-labs-pagination-next-button" icon="tier1:chevron-right" @click="${this._onNextClicked}" text="${this.localize('components:pagination:nextPage')}" ?disabled=${this._isLastPage()}></d2l-button-icon>
+			</div>
+
+			${this.showItemCountSelect ? html`
+				<select
```

Comment:
Is there any benefit to select or the options having an `id`?

Reply:
Not really, since there's only a single select, it probably isn't worth having an id for it. On the other side, since the options are passed in by the user, I don't want to risk the array having duplicates and causing things to fail due to there being two components with the same ID (since they're meant to be unique)